### PR TITLE
Sourcery Starbot ⭐ refactored felipmartins/fastapi-swapi

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>main</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/fastapi-swapi main
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: swapi/main.py [hunk 1]
```diff
@@ -68,20 +68,18 @@ async def search_planets(name: str = None, gravity: str = None):
             planets = session.exec(select(Planet).where(Planet.name == name)).all()
         elif not name and gravity:
             planets = session.exec(select(Planet).where(Planet.gravity == gravity)).all()
-        elif name and gravity:
+        elif name:
             planets = session.exec(select(Planet).where(Planet.name == name).where(Planet.gravity == gravity)).all()
         else:
             planets = session.exec(select(Planet)).all()
-    
+
```
Comment (SourceryAI) [inline swapi/main.py:75]: Function `search_planets` refactored with the following changes:

- Remove redundant conditional ([`remove-redundant-if`](https://docs.sourcery.ai/Reference/Built-in-Rules/refactorings/remove-redundant-if/))

File: swapi/main.py [hunk 2]
```diff
@@ -68,20 +68,18 @@ async def search_planets(name: str = None, gravity: str = None):
...
         return create_response(planets)
 
 
 @app.get("/api/planets/{id}", tags=["planets"], response_model=PlanetRead)
 async def list_planet_by_id(id: int):
     with get_session() as session:
-        planet = session.exec(select(Planet).where(Planet.id == id)).one()
-
-        return planet
+        return session.exec(select(Planet).where(Planet.id == id)).one()
```
Comment (SourceryAI) [inline swapi/main.py:82]: Function `list_planet_by_id` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/Reference/Built-in-Rules/refactorings/inline-immediately-returned-variable/))

File: swapi/main.py [hunk 3]
```diff
@@ -105,6 +103,4 @@ async def create_people(people: PeopleCreate):
 @app.get("/api/people/{id}", tags=["people"], response_model=PeopleRead)
 async def list_people_by_id(id: int):
     with get_session() as session:
-        people = session.exec(select(People).where(People.id == id)).one()
-
-        return people
+        return session.exec(select(People).where(People.id == id)).one()
```
Comment (SourceryAI) [inline swapi/main.py:106]: Function `list_people_by_id` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/Reference/Built-in-Rules/refactorings/inline-immediately-returned-variable/))

File: swapi/main.py (hunk 1 above)
Comment (ipfalvim) [reply swapi/main.py:75]: porque seria redundante? caso falte `gravity` nao iria quebrar a linha 72?

Comment (vbuxbaum) [reply swapi/main.py:75]: @ipfalvim porque as condicionais anteriores já possuem todas as variações de expressões de lógica booleana entre `name` e `gravity`

Comment (ipfalvim) [reply swapi/main.py:75]: top, ajuda demais então nessas micro otimizações @vbuxbaum

**Pull request merged**